src/daft-io/src/google_cloud.rs

```diff
@@ -393,10 +452,48 @@ impl GCSSource {
         if config.project_id.is_some() {
             client_config.project_id.clone_from(&config.project_id);
         }
+        client_config.http = Some({
+            use reqwest_middleware::ClientBuilder;
+            use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
+            use retry_policies::Jitter;
+            let retry_policy = ExponentialBackoff::builder()
+                .base(2)
+                .jitter(Jitter::Bounded)
+                .retry_bounds(
+                    Duration::from_millis(config.retry_initial_backoff_ms),
+                    Duration::from_secs(60),
+                )
+                .build_with_max_retries(config.num_tries);
+
+            let base_client = reqwest_middleware::reqwest::ClientBuilder::default()
+                .connect_timeout(Duration::from_millis(config.connect_timeout_ms))
+                .read_timeout(Duration::from_millis(config.read_timeout_ms))
+                .pool_idle_timeout(Duration::from_secs(60))
+                .pool_max_idle_per_host(70)
```

Comment:
Why 70? How many connections does it create anyways

Reply:
This is for idle connections for connection reuse. this is the default for many AWS SDKS